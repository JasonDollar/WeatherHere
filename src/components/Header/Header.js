import React from 'react'
import propTypes from 'prop-types'
import classes from './Header.module.scss'

import Options from '../Options/Options'

const Header = ({
  text, searchValue, onSearchFormSubmit, changeLanguage, onButtonClick, onInputChange,
}) => (
  <div className={classes.container}>
    <h1 className={classes.header}>{text.title}</h1> 
    <form onSubmit={onSearchFormSubmit}>
      <input type="text" onChange={onInputChange} value={searchValue} />
      <button type="submit">{text.search}</button>
    </form>
    <Options changeLanguage={changeLanguage} />
    <button onClick={onButtonClick}>Geolocatipon</button>
  </div>
)

export default Header

Header.propTypes = {
  text: propTypes.objectOf(propTypes.string),
}