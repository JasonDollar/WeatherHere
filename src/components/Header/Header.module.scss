.container {
  padding: 1rem 0;

  @media only screen and (min-width: 577px) {

    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
    padding-bottom: 0;

    &>* {
      margin: 0 .5rem;
    }
  }
  @media only screen and (min-width: 806px) {
    padding-bottom: 1rem;
  }
}

.header {
  margin: 0;
  margin-left: 1.5rem;
  color: var(--color-text-secondary);
  // padding-bottom: 1rem;
}