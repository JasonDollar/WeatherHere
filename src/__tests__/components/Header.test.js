import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Header from '../../components/Header/Header'

test('Header component matches snapshot', () => {
  const wrapper = shallow(<Header />)

  expect(toJSON(wrapper)).toMatchSnapshot() 
})