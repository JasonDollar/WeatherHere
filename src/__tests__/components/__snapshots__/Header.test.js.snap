// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<header
  className="container"
>
  <h1
    className="header"
  >
    WeatherHere
  </h1>
</header>
`;
